src/components/Checkout/AutoDepositUpsell.vue

```diff
@@ -0,0 +1,269 @@
+						class="tw-w-5 tw-h-5 tw-mr-1.5"
+						alt="Clock icon"
+					>
+					<div>
+						<h3 class="tw-pb-1">
+							Automatic
+						</h3>
+						<p>Each month, the amount you’ve set will automatically be added to your Kiva account.</p>
+					</div>
+				</div>
+				<div class="tw-flex tw-mx-auto md:tw-mx-4 tw-mb-3.5 md:tw-mb-2 md:tw-flex-1" style="max-width: 295px;">
+					<img
+						:src="imageRequire(`./heart-icon-encircled.svg`)"
+						class="tw-w-5 tw-h-5 tw-mr-1.5"
+						alt="Arrow around a heart icon"
+					>
+					<div>
+						<h3 class="tw-pb-1">
+							Impactful
+						</h3>
+						<p>Once you have enough for a loan, we’ll send you an email reminder to pick a borrower.</p>
+					</div>
+				</div>
+			</div>
+		</kv-lightbox>
+	</div>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+import experimentAssignmentQuery from '@/graphql/query/experimentAssignment.graphql';
+import experimentVersionFragment from '@/graphql/fragments/experimentVersion.graphql';
+import KvLightbox from '~/@kiva/kv-components/vue/KvLightbox';
+// TODO: Swap in KvSwitch after reactive bugs are fixed
+// import KvSwitch from '~/@kiva/kv-components/vue/KvSwitch';
+import KvCheckbox from '~/@kiva/kv-components/vue/KvCheckbox';
+
+const imageRequire = require.context('@/assets/images/kiva-classic-illustrations/', true);
+
+const cookieName = 'kv-show-ad-signup';
+
+const eligibilityCheckQuery = gql`query autoDepositEligibilityQuery {
+	general {
+		autoDepositUpsellSeting: uiExperimentSetting(key: "checkout_ad_upsell") {
+			key
+			value
+		}
+	}
+	my {
+		userAccount{
+			id
+		}
+		autoDeposit {
+			id
+			isSubscriber
+			isOnetime
+		}
+		subscriptions {
+			values {
+				id
+			}
+		}
+	}
+	mySubscriptions(includeDisabled: false) {
+		values {
+			id
+			enabled
+			category {
+				id
+				subscriptionType
+			}
+		}
+	}
+}`;
+
+export default {
+	name: 'AutoDepositUpsell',
+	inject: ['apollo', 'cookieStore'],
+	components: {
+		KvCheckbox,
+		KvLightbox,
+		// KvSwitch,
+	},
+	props: {
+		myId: {
+			type: Number,
+			default: 0
+		}
+	},
+	data() {
+		return {
+			autoDepositUpsellExpVersion: null,
+			adOptIn: false,
+			imageRequire,
+			lightboxVisible: false,
+			userAccountId: null,
+		};
+	},
+	computed: {
+		experimentVisible() {
+			if (this.autoDepositUpsellExpVersion === 'b') {
+				return true;
+			}
+			return false;
+		},
+		userId() {
+			if (this.myId !== 0) {
+				return this.myId;
+			}
+			if (this.userAccountId !== null) {
+				return this.userAccountId;
+			}
+			return null;
+		}
+	},
+	apollo: {
+		preFetch(config, client) {
+			return client.query({
+				query: eligibilityCheckQuery
+			}).then(({ data }) => {
+				const isLoggedIn = data?.my?.userAccountId?.id !== null;
+				const hasAutoDeposit = data?.my?.autoDeposit !== null;
+				const hasLegacySubs = data?.my?.subscriptions?.values?.length !== 0;
+				const hasModernSub = data?.mySubscriptions?.values.length !== 0;
```

Comment:
Just an FYI we are not checking for modern subs anywhere else yet. For example a cause user can go to the  AD page and sign up. 

We are checking various eligibilities in several places now. It would be nice to have some global utility functions that just aggregate all this stuff. `isEligibleForAutoDeposit`

Reply:
This is potentially problematic as it would cause "dual" subscriptions and probably get their email settings in a really weird place. Could you please create a ticket to add these checks on AD/MG signups?